includes/functions/content-sync.php

```diff
@@ -146,10 +146,33 @@ function get_post_data( $post ) {
 	 */
 	$type = apply_filters( 'sophi_post_data_type', get_post_format( $post ), $post );
 
-	if ( ! in_array( $type, [ 'video', 'audio', 'image'], true ) ) {
+	if ( ! in_array( $type, [ 'video', 'audio', 'image' ], true ) ) {
 		$type = 'article';
 	}
 
+	$canonical_url = wp_get_canonical_url( $post );
+
+	// Support Yoast SEO canonical URL.
+	if ( class_exists( 'WPSEO_Options' ) ) {
+		$yoast_canonical = get_post_meta( $post->ID, '_yoast_wpseo_canonical', true );
+		if ( $yoast_canonical ) {
+			$canonical_url = $yoast_canonical;
+		}
+	}
+
+	/**
+	 * Filter canonical URL of the given post.
+	 *
+	 * @since 1.0.4
+	 * @hook sophi_post_canonical_url
+	 *
+	 * @param {string}  $canonical_url Canonical URL of given post.
+	 * @param {WP_Post} $post WP_Post object.
+	 *
+	 * @return {string} Canonical URL.
+	 */
+	$canonical_url = apply_filters( 'sophi_post_canonical_url', $canonical_url, $post );
```

Comment:
As discussed, I'm fine if we want to leave this filter but there's probably not a great need to have this, since there are other places this data can be filtered. If we want to remove this to simplify things, that's good with me.